private/model/cli/gen-api/main.go

```diff
@@ -112,92 +45,102 @@ func newGenerateInfo(modelFile, svcPath, svcImportPath string) *generateInfo {
-	flag.StringVar(&svcImportPath, "svc-import-path", "github.com/aws/aws-sdk-go/service", "namespace to generate service client Go code import path under")
+	var svcPath, svcImportPath string
+	flag.StringVar(&svcPath, "path", "service",
+		"The `path` to generate service clients in to.",
+	)
+	flag.StringVar(&svcImportPath, "svc-import-path",
+		"github.com/aws/aws-sdk-go/service",
+		"The Go `import path` to generate client to be under.",
+	)
+	flag.Usage = usage
 	flag.Parse()
-	api.Bootstrap()
 
 	if len(os.Getenv("AWS_SDK_CODEGEN_DEBUG")) != 0 {
 		api.LogDebug(os.Stdout)
 	}
 
-	files := []string{}
-	for i := 0; i < flag.NArg(); i++ {
-		file := flag.Arg(i)
-		if strings.Contains(file, "*") {
-			paths, _ := filepath.Glob(file)
-			files = append(files, paths...)
-		} else {
-			files = append(files, file)
-		}
+	modelPaths, err := api.ExpandModelGlobPath(flag.Args()...)
+	if err != nil {
+		fmt.Fprintln(os.Stderr, "failed to glob file pattern", err)
+		os.Exit(1)
 	}
 
-	for svcName := range excludeServices {
-		if strings.Contains(os.Getenv("SERVICES"), svcName) {
-			fmt.Fprintf(os.Stderr, "Service %s is not supported\n", svcName)
-			os.Exit(1)
-		}
+	apis, err := api.LoadAPIs(modelPaths, svcImportPath)
+	if err != nil {
+		fmt.Fprintln(os.Stderr, "failed to load API models", err)
+		os.Exit(1)
+	}
+	if len(apis) == 0 {
+		fmt.Fprintf(os.Stderr, "expected to load models, but found none")
+		os.Exit(1)
 	}
 
-	sort.Strings(files)
-
-	// Remove old API versions from list
-	m := map[string]bool{}
-	// caches paths to ensure we are not overriding previously generated
-	// code.
-	servicePaths := map[string]struct{}{}
-
-	for i := range files {
-		idx := len(files) - 1 - i
-		parts := strings.Split(files[idx], string(filepath.Separator))
-		svc := parts[len(parts)-3] // service name is 2nd-to-last component
-
-		if m[svc] {
-			files[idx] = "" // wipe this one out if we already saw the service
+	if v := os.Getenv("SERVICES"); len(v) != 0 {
+		svcs := strings.Split(v, ",")
+		for pkgName, a := range apis {
+			var found bool
+			for _, include := range svcs {
+				if a.PackageName() == include {
+					found = true
+					break
+				}
+			}
+			if !found {
+				delete(apis, pkgName)
+			}
 		}
-		m[svc] = true
 	}
 
-	wg := sync.WaitGroup{}
-	for i := range files {
-		filename := files[i]
-		if filename == "" { // empty file
+	var wg sync.WaitGroup
+	servicePaths := map[string]struct{}{}
+	for _, a := range apis {
+		if _, ok := excludeServices[a.PackageName()]; ok {
 			continue
 		}
 
-		genInfo := newGenerateInfo(filename, svcPath, svcImportPath)
-		if genInfo == nil {
-			continue
-		}
-		if _, ok := excludeServices[genInfo.API.PackageName()]; ok {
-			// Skip services not yet supported.
-			continue
-		}
+		// Create the output path for the model.
+		pkgDir := filepath.Join(svcPath, a.PackageName())
+		os.MkdirAll(pkgDir, 0775)
```

Comment:
this line technically isn't needed.

Reply:
good point in that case we can probably drop it.